recovery/src/main/java/com/kin/ecosystem/recovery/backup/presenter/BackupPresenter.java

```diff
@@ -1,9 +1,13 @@
 package com.kin.ecosystem.recovery.backup.presenter;
 
+import android.os.Bundle;
 import com.kin.ecosystem.recovery.backup.view.BackupNextStepListener;
 import com.kin.ecosystem.recovery.backup.view.BackupView;
 import com.kin.ecosystem.recovery.base.BasePresenter;
 
 public interface BackupPresenter extends BasePresenter<BackupView>, BackupNextStepListener {
 
+	void onSaveInstanceState(Bundle outState);
+
+	void saveKeyData(String key);
```

Comment:
setKey could be better - the save is internal detail

Reply:
Consider to provide API for get/set  key and step explicitly within the presenter. The view will be able to set and also get the current value in the presenter. The Activity/Fragment responsibility to persist (by calling get on these property)  or restore it (by set) when needed